gninasrc/lib/cnn_scorer.cpp

```diff
@@ -358,8 +484,28 @@ float CNNScorer::score(model& m, bool compute_gradient, float& affinity,
     if (compute_gradient || cnnopts.outputxyz) {
 
       net->Backward();
+
+      // Get ligand gradient
       mgrid->getLigandGradient(0, gradient);
+
+      CHECK_EQ(gradient.size(), ligand_coords.size());
+
+      // Get receptor gradient
```

Comment:
Extraction of the receptor gradient for flexible atoms and the concatenation to the ligand gradient can be extracted in an utility function returning the full gradient.